docs/en/integrations/language-clients/java/index.md

```diff
@@ -5,126 +5,250 @@ keywords: [clickhouse, java, jdbc, client, integrate, r2dbc]
+    } catch (Exception e) {
+        log.error("Failed to write JSONEachRow data", e);
+        throw new RuntimeException(e);
+    }
+}
 
-```java
-ClickHouseNode server = ClickHouseNode.of("http://localhost:8123/default?compress=0");
 ```
 
-### Compression
+### Query API
 
-The client will by default use LZ4 compression, which requires this dependency:
+| Method                                                             | Description |
+|--------------------------------------------------------------------|-------------|
+| `CompletableFuture<QueryResponse> query(String sqlQuery, QuerySettings settings)` | Sends `sqlQuery` as is. Response format is set by query settings. `QueryResponse` will hold a reference to the response stream what should be consumer by a reader for supportig format | 
+| `CompletableFuture<QueryResponse> query(String sqlQuery, Map<String, Object> queryParams, QuerySettings settings)` | Sends `sqlQuery` as is. Additionally will send query parameter so server can comple SQL expression. | 
+| `ClickHouseBinaryFormatReader newBinaryFormatReader(QueryResponse response, TableSchema schema)` | Creates a reader for `RowBinary*` and `Native*`. Table schema is required for `RowBinaryWithNames` and `RowBinary` formats. 'Native' and `RowBinaryWithNamesAndTypes` has the schema within data stream. |
+| `List<GenericRecord> queryAll(String sqlQuery)` | Queries a data in `RowBinaryWithNamesAndTypes` format. Returns result as a collection. Read performance is the same as with reader but more memory required at a time to keep whole dataset. |
+
+**QuerySettings**
+
+| Configuration Method                       | Default Value | Description | 
+|--------------------------------------------|---------------|-------------|
+| `QuerySettings setQueryId(String queryId)` |               | Sets query ID that will be assigned to the operation | 
+| `QuerySettings setFormat(ClickHouseFormat format)` | `RowBinaryWithNamesAndTypes` | Sets response format. See `RowBinaryWithNamesAndTypes` for the full list. |
+| `QuerySettings setMaxExecutionTime(Integer maxExecutionTime)` |        | Sets operation execution time on server. Will not affect read timeout. | 
+| `QuerySettings waitEndOfQuery(Boolean waitEndOfQuery)` | `false` | Requests the server to wait for the and of the query before sending response. | 
+| `QuerySettings setUseServerTimeZone(Boolean useServerTimeZone)` | `true` | Server timezone (see client config) will be used to parse date/time types in the result of an operation. |
+| `QuerySettings setUseTimeZone(String timeZone)` |    | Requests server to use `timeZone` for time conversion. See (`session_timezone`)[https://clickhouse.com/docs/en/operations/settings/settings#session_timezone] |
+
+
+
+### Examples 
+
+- Example code is available in [repo](https://github.com/ClickHouse/clickhouse-java/tree/main/examples/client-v2)
```

Comment:
Adding a link with examples to each section (insert, query, etc.) would improve UX